version: "3.9"

services:

  flask:
    build: .
    command: "gunicorn -w 2 main:app --bind 0.0.0.0:${FLASK_INTERNAL_PORT}"
    expose:
      - "${FLASK_INTERNAL_PORT}"
    restart: on-failure
    volumes:
      - ./.env:/app/src/.env
      - ./logs/nginx/json-logs.json:/app/src/ugc_service.log
    env_file:
      - .env
    networks:
      - elk_default

  nginx:
    image: "nginx:1.25.3"
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf:ro
      - ./logs/nginx/:/var/log/nginx/
      - ./configs:/etc/nginx/conf.d:ro
      - ./data/html:/data/html:ro
    depends_on:
      - flask
    ports:
      - "80:80"
    networks:
      - elk_default	

  filebeat:
    image: elastic/filebeat:8.10.2
    command: filebeat -e -strict.perms=false
    volumes:
      - ./logs/nginx:/var/log/nginx:ro
      - ./deploy/filebeat.yml:/usr/share/filebeat/filebeat.yml
    networks:
      - elk_default

networks:
  elk_default:
    name: elk_name
